modules/stroeerCoreBidAdapter.js

```diff
@@ -0,0 +1,249 @@
+        }
+      }
+    };
+
+    function isBanner(bidReq) {
+      return (!bidReq.mediaTypes && !bidReq.mediaType) ||
+        (bidReq.mediaTypes && bidReq.mediaTypes.banner) ||
+        bidReq.mediaType === BANNER;
+    }
+
+    validators.push(createValidator((bidReq) => isBanner(bidReq),
+      bidReq => `bid request ${bidReq.bidId} is not a banner`));
+    validators.push(createValidator((bidReq) => typeof bidReq.params === 'object',
+      bidReq => `bid request ${bidReq.bidId} does not have custom params`));
+    validators.push(createValidator((bidReq) => utils.isStr(bidReq.params.sid),
+      bidReq => `bid request ${bidReq.bidId} does not have a sid string field`));
+    validators.push(createValidator((bidReq) => bidReq.params.ssat === undefined || [1, 2].indexOf(bidReq.params.ssat) > -1,
+      bidReq => `bid request ${bidReq.bidId} does not have a valid ssat value (must be 1 or 2)`));
+
+    return function (bidRequest) {
+      return validators.every(f => f(bidRequest));
+    }
+  }()),
+
+  buildRequests: function (validBidRequests = [], bidderRequest) {
+    const anyBid = bidderRequest.bids[0];
+
+    const bidRequestWithSsat = find(validBidRequests, bidRequest => bidRequest.params.ssat);
+    const bidRequestWithYl2 = find(validBidRequests, bidRequest => bidRequest.params.yl2);
+
+    const payload = {
+      id: bidderRequest.auctionId,
+      bids: [],
+      ref: getTopWindowReferrer(),
+      ssl: isSecureWindow(),
+      mpa: isMainPageAccessible(),
+      timeout: bidderRequest.timeout - (Date.now() - bidderRequest.auctionStart),
+      ssat: bidRequestWithSsat ? bidRequestWithSsat.params.ssat : 2, // TODO
+      yl2: bidRequestWithYl2 ? bidRequestWithYl2.params.yl2 : (localStorage.sdgYieldtest === '1'), // TODO
+      ab: utils.getWindowSelf()['yieldlove_ab'] // TODO
+    };
+
+    const userIds = anyBid.userId;
+
+    if (!utils.isEmpty(userIds)) {
+      payload.user = {
+        euids: userIds
+      };
+    }
+
+    const gdprConsent = bidderRequest.gdprConsent;
+
+    if (gdprConsent && gdprConsent.consentString != null && gdprConsent.gdprApplies != null) {
+      payload.gdpr = {
+        consent: bidderRequest.gdprConsent.consentString, applies: bidderRequest.gdprConsent.gdprApplies
+      };
+    }
+
+    function bidSizes(bid) {
+      return utils.deepAccess(bid, 'mediaTypes.banner.sizes') || bid.sizes /* for prebid < 3 */ || [];
+    }
+
+    validBidRequests.forEach(bid => {
+      payload.bids.push({
+        bid: bid.bidId, sid: bid.params.sid, siz: bidSizes(bid), viz: elementInView(bid.adUnitCode)
+      });
+    });
+
+    return {
+      method: 'POST', url: buildUrl(anyBid.params), data: payload
+    }
+  },
+
+  interpretResponse: function (serverResponse) {
+    const bids = [];
+
+    if (serverResponse.body && typeof serverResponse.body === 'object') {
+      // TODO: do we need this ajax call?
+      if (serverResponse.body.tep) {
+        ajax(serverResponse.body.tep, () => {
```

Comment:
What about this call? We think we cannot execute ajax calls in this phase. Can we get rid of it?

Reply:
Yes that can be removed. It is used for tracking delivery.